File: lib/scraper/text_scraper.py
import time

from .join_scraper import JoinScraper
from ..driver.support.driver_support import DriverSupport


class TextScraper(JoinScraper):
    """Scraper object focused on text content"""

    def __init__(self, driver, opts, nav_info, file_filter=None, sleep=False):
        """Initializes TextScraper object

       :param WebDriver driver: Selenium Webdriver
        :param Opts opts: Opts class
        :param NavInfo nav_info: NavInfo object
        :param generic.Generic file_filter: Filter object to use for files
        :param bool sleep: stops the program before scraping
        """
        super().__init__(driver, opts, nav_info, file_filter, sleep)

    def _join_loop(self, element, directory) -> None:
        link = element.text.strip()
        self._join_links(directory, link)

    def scrape_items(self) -> list:
        if self._sleep:
            time.sleep(4)
        return DriverSupport.get_elements_all(self._driver, self._opts,
                                              self.nav_info.css_name,
                                              self.nav_info.wait_err_message)

File: app/src/main/java/xyz/zimtools/zyod/scrapers/TextScraper.java
package xyz.zimtools.zyod.scrapers;

import org.openqa.selenium.WebElement;
import org.openqa.selenium.remote.RemoteWebDriver;
import xyz.zimtools.zyod.args.Args;
import xyz.zimtools.zyod.assets.Directory;
import xyz.zimtools.zyod.assets.info.NavInfo;
import xyz.zimtools.zyod.scrapers.filters.ScrapeFilter;
import xyz.zimtools.zyod.support.NavSupport;

import java.util.List;

/**
 * Scrape ODs using the {@link WebElement}'s text content.
 */
public class TextScraper extends AttributeScraper {

    public TextScraper(RemoteWebDriver driver, Args args, NavInfo navInfo, ScrapeFilter filter) {
        super(driver, args, navInfo, filter);
    }

    public TextScraper(RemoteWebDriver driver, Args args, NavInfo navInfo) {
        super(driver, args, navInfo);
    }

    /**
     * Scrape for {@link WebElement}s using CSS name
     *
     * @return list of scraped WebElements
     */
    @Override
    public List<WebElement> scrapeItems() {
        this.pause();
        return NavSupport.getElements(this.driver, this.navInfo.getCssFileName());
    }

    /**
     * Join text content found in {@link WebElement} to the current URL.
     *
     * @param element   scraped WebElement
     * @param parentDir the current parent {@link Directory}
     */
    @Override
    protected void joinLoop(WebElement element, Directory parentDir) {
        String text = element.getText().strip();
        this.joinLinks(text, parentDir);
    }
}